examples/models/image_classification/TfFeedForward.py

```diff
@@ -8,110 +8,89 @@
     for simple image classification
     '''
-
-    def get_knob_config(self):
-        epochs_range = [3, 100]
-        hidden_layer_count_range = [1, 8]
-        
-        if APP_MODE == 'DEV':
-            print('WARNING: In DEV mode, `epochs` is set to 3 and `hidden_layer_count` is set to 2.')
-            epochs_range = [3, 3]
-            hidden_layer_count_range = [2, 2]
-
+    @staticmethod
+    def get_knob_config():
         return {
-            'knobs': {
-                'epochs': {
-                    'type': 'int',
-                    'range': epochs_range
-                },
-                'hidden_layer_count': {
-                    'type': 'int',
-                    'range': hidden_layer_count_range
-                },
-                'hidden_layer_units': {
-                    'type': 'int',
-                    'range': [2, 128]
-                },
-                'learning_rate': {
-                    'type': 'float_exp',
-                    'range': [1e-5, 1e-1]
-                },
-                'batch_size': {
-                    'type': 'int_cat',
-                    'values': [16, 32, 64, 128]
-                },
-                'image_size': {
-                    'type': 'int_cat',
-                    'values': [8, 16, 32]
-                }
-            }
+            'epochs': IntegerKnob(3, 10 if APP_MODE != 'DEV' else 3),
+            'hidden_layer_count': IntegerKnob(1, 8 if APP_MODE != 'DEV' else 2),
+            'hidden_layer_units': IntegerKnob(2, 128),
+            'learning_rate': FloatKnob(1e-5, 1e-1, is_exp=True),
+            'batch_size': CategoricalKnob([16, 32, 64, 128]),
+            'image_size': FixedKnob(32)
         }
 
-    def init(self, knobs):
-        self._batch_size = knobs.get('batch_size')
-        self._hidden_layer_units = knobs.get('hidden_layer_units')
-        self._hidden_layer_count = knobs.get('hidden_layer_count')
-        self._learning_rate = knobs.get('learning_rate')
-        self._epochs = knobs.get('epochs')
-        self._image_size = knobs.get('image_size')
-
+    def __init__(self, **knobs):
+        super().__init__(**knobs)
+        self._knobs = knobs
         self._graph = tf.Graph()
         config = tf.ConfigProto()
         config.gpu_options.allow_growth = True
         self._sess = tf.Session(graph=self._graph, config=config)
-        self._define_plots()
         
     def train(self, dataset_uri):
-        dataset = self.utils.load_dataset_of_image_files(dataset_uri, image_size=[self._image_size, self._image_size])
+        im_sz = self._knobs.get('image_size')
+        bs = self._knobs.get('batch_size')
+        ep = self._knobs.get('epochs')
+
+        logger.log('Available devices: {}'.format(str(device_lib.list_local_devices())))
+
+        # Define 2 plots: Loss against time, loss against epochs
+        logger.define_loss_plot()
+        logger.define_plot('Loss Over Time', ['loss'])
```

Comment:
timestamp is always recorded.
can we always plot 2 figures including one with x-axis being the timestamp?